zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -888,18 +886,16 @@ public void removeRepository(String id) throws IOException {
   }
 
   public void removeNoteInterpreterSettingBinding(String user, String noteId) throws IOException {
-    synchronized (interpreterSettings) {
```

Comment:
This synchronize block is for protecting `interpreterBindings`, and i think we still need for synchronize block for `interpreterBindings`.

However, `this.removeInterpreterForNote()` doesn't have to be protected by synchronize block.
So i think code can be look like

```
List<String> settingIds;
synchronize (interpreterSettings) {
   settingIds = interpreterBindings.containsKey(...) ? interpreterBindings.remove(...) : ...
}

for (String settingId : settingIds) {
  ...
  this.removeInterpreterForNote(...);
  ...
}
```